src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -829,108 +830,132 @@ subroutine init_atm_static(mesh, dims, configs)
 
-    do jTileStart = 1,02401,ny-6
-       jTileEnd = jTileStart + ny - 1 - 6
-
-       do iTileStart=1,06001,nx-6
-          iTileEnd = iTileStart + nx - 1 - 6
-          write(fname,'(a,i5.5,a1,i5.5,a1,i5.5,a1,i5.5)') trim(geog_data_path)//trim(geog_sub_path), &
-                       iTileStart,'-',iTileEnd,'.',jTileStart,'-',jTileEnd
-          call mpas_log_write(trim(fname))
-          call mpas_f_to_c_string(fname, c_fname)
+    allocate(nhs(nCells))
+    snoalb(:) = 0.0
+    nhs(:) = 0
+    fillval = 0.0
+    missing_value = real(scalefactor * missing_value, kind=R4KIND)
+
+    do iCell = 1, nCells
+        if (nhs(iCell) == 0) then
+            tile => null()
+            ierr = mgr % get_tile(latCell(iCell), lonCell(iCell), tile)
+            if (ierr /= 0 .or. .not. associated(tile)) then
+                call mpas_log_write('Could not get tile that contained cell $i', intArgs=(/iCell/), messageType=MPAS_LOG_CRIT)
+            end if
 
-          call read_geogrid(c_fname,rarray_ptr,nx,ny,nz,isigned,endian, &
-                            wordsize,istatus)
-          call init_atm_check_read_error(istatus, fname)
-          rarray(:,:,:) = rarray(:,:,:) * scalefactor
+            ierr = mgr % push_tile(tile)
+            if (ierr /= 0) then
+                call mpas_log_write("Error pushing this tile onto the stack: "//trim(tile%fname), messageType=MPAS_LOG_CRIT)
+            end if
+        end if
 
-          iPoint = 1
-          do j=supersample_fac * 3 + 1, supersample_fac * (ny-3)
-          do i=supersample_fac * 3 + 1, supersample_fac * (nx-3)
-             ii = (i - 1) / supersample_fac + 1
-             jj = (j - 1) / supersample_fac + 1
+        do while (.not. mgr % is_stack_empty())
+            tile => mgr % pop_tile()
 
-             lat_pt = start_lat - (supersample_fac*(jTileStart-1) + j - (supersample_fac*3+1)) * 0.05 / supersample_fac
-             lon_pt = start_lon + (supersample_fac*(iTileStart-1) + i - (supersample_fac*3+1)) * 0.05 / supersample_fac
-             lat_pt = lat_pt * PI / 180.0
-             lon_pt = lon_pt * PI / 180.0
+            if (tile % is_processed) then
+                cycle
+            end if
 
-             iPoint = nearest_cell(lat_pt,lon_pt,iPoint,nCells,maxEdges, &
-                                   nEdgesOnCell,cellsOnCell, &
-                                   latCell,lonCell)
-             if (rarray(ii,jj,1) /= msgval) then
+            call mpas_log_write('Processing tile: '//trim(tile % fname))
+
+            all_pixels_mapped_to_halo_cells = .true.
+
+            do j = supersample_fac * tile_bdr + 1, supersample_fac * (tile_ny + tile_bdr), 1
+                do i = supersample_fac * tile_bdr + 1, supersample_fac * (tile_nx + tile_bdr), 1
+
+                    ii = (i - 1) / supersample_fac + 1
+                    jj = (j - 1) / supersample_fac + 1
+
+                    call mgr % tile_to_latlon(tile, j, i, lat_pt, lon_pt, supersample_fac)
```

Comment:
When `supersample_fac` != 1, is calling `tile_to_latlon` with `i`, `j`, and `supersample_fac` equivalent to calling the routine with `ii`, `jj`, and `1`? In other words, if we "un-super-sample" the pixel coordinates at this level in the code, is there any need to handle a supersampling factor in the `tile_to_latlon` routine?

Reply:
I don't think the two are equivalent. `supersample_fac` needs to be passed into `tile_to_latlon` so that we can divide the pixel into subpixels: `dx / supersample`. 

``` Fortran
lat = known_lat + real(j - (supersample_lcl * tile_bdr + 1) + (supersample_lcl * (tile % y - 1)), kind=RKIND) * dy &
                               / supersample_lcl
lon = known_lon + real(i - (supersample_lcl * tile_bdr + 1) + (supersample_lcl * (tile % x - 1)), kind=RKIND) * dx &
                              / supersample_lcl
```

Here is a toy program which I made that demonstrates that the two aren't equivalent:

https://gist.github.com/MiCurry/a237abff4ec5596ec4c754e38e0f4a99

```
 not equal
 First:           11          10   89.9749985      -179.958344               3
 Second:            4           4   89.9749985      -179.975006               1
```

Does that seem correct/right to you?

Reply:
I think the result printed in the `Second:` line is correct. With a supersampling ratio of 3, the first 3x3 block of sub-pixels should all map to the same latitude and longitude, i.e., the latitude and longitude of the first full pixel.

Reply:
I thought the idea behind supersample was so that the sub-pixels did not map to the same location as the full pixel. So for instance: With a supersample of 2, if an edge of two cells, c1 and c2, divided a full pixel equally in half in the x-direction, then the first half of the sub-pixels would be mapped to c1 and the second half of sub-pixels would be mapped to c2. Is that assumption incorrect?

Reply:
Sorry for the confusion -- you're correct: when computing the latitude and longitude of sub-pixels, the computed location should vary! I somehow got stuck thinking about the reverse problem: that different latitudes and longitudes should map to the same (j,i) location.